src/sparse_blas/backends/mkl_common/mkl_handles.hpp

```diff
@@ -0,0 +1,76 @@
+/***************************************************************************
+*  Copyright (C) Codeplay Software Limited
+*  Licensed under the Apache License, Version 2.0 (the "License");
+*  you may not use this file except in compliance with the License.
+*  You may obtain a copy of the License at
+*
+*      http://www.apache.org/licenses/LICENSE-2.0
+*
+*  For your convenience, a copy of the License has been included in this
+*  repository.
+*
+*  Unless required by applicable law or agreed to in writing, software
+*  distributed under the License is distributed on an "AS IS" BASIS,
+*  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+*  See the License for the specific language governing permissions and
+*  limitations under the License.
+*
+**************************************************************************/
+
+#ifndef _ONEMKL_SRC_SPARSE_BLAS_BACKENDS_MKL_COMMON_MKL_HANDLES_HPP_
+#define _ONEMKL_SRC_SPARSE_BLAS_BACKENDS_MKL_COMMON_MKL_HANDLES_HPP_
+
+// MKLCPU and MKLGPU backends include
+// This include defines its own oneapi::mkl::sparse namespace with some of the types that are used here: matrix_handle_t, index_base, transpose, uplo, diag.
+#include <oneapi/mkl/spblas.hpp>
+
+#include "sparse_blas/generic_container.hpp"
+
+namespace oneapi::mkl::sparse {
+
+// Complete the definition of incomplete types dense_vector_handle and dense_matrix_handle as there is no conflict with the oneMKL backends.
+
+struct dense_vector_handle : public detail::generic_dense_vector_handle<void*> {
+    template <typename T>
```

Comment:
Do we inherit from oneMKL Interfaces' `generic_dense_vector_handle` right now because Intel oneMKL does not have a `dense_vector_handle` in it? If/when Intel oneMKL adds it, would we remove this?

Reply:
Yes, exactly. This struct is just here to complete the definition of the opaque types. I wish I could do a `using dense_vector_handle = detail::generic_dense_vector_handle<void*>` but that's not valid to complete the definition.